akka-stream-typed/src/test/scala/docs/akka/stream/typed/ActorSourceSinkExample.scala

```diff
@@ -42,6 +42,71 @@ object ActorSourceSinkExample {
     // #actor-source-ref
   }
 
+  {
+
+    implicit val system: ActorSystem[_] = ???
+
+    // #actor-source-with-backpressure
+    import akka.actor.typed.ActorRef
+    import akka.stream.CompletionStrategy
+    import akka.stream.scaladsl.{ Sink, Source }
+    import akka.stream.typed.scaladsl.ActorSource
+
+    case object Ack
+
+    trait Protocol
+    case class Message(msg: String) extends Protocol
+    case object Complete extends Protocol
+    case class Fail(ex: Exception) extends Protocol
```

Comment:
Since the materialized source actorref cannot validate these and choose how to act, but must always accept them, what about calling the top type `Event` and aligning naming of complete to sound less like commands, `End` and `Failure` for example. Pehaps also call `Message` `Element` instead as the purpose is to turn it into an element in the stream?